src/main/java/co/alphavantage/input/exchange/Currency.java

```diff
@@ -0,0 +1,62 @@
+package co.alphavantage.input.exchange;
+
+import co.alphavantage.input.ApiParameter;
+
+/**
+* Interval parameter, i.e interval=15min
+*/
+public class Currency {

```

Comment:
This looks a bit complicated to me, why not simply have a FromCurrency and a ToCurrency?

Reply:
I believe he does it to favour code reusability and avoid duplication of currency codes.

Reply:
Right, but maybe the From and To should instead extend FromTo that keeps the value. Right now I dont see the use of the Currency class since it keeps a instance of both From and To. I propose to get rid of the Currency class and let the currency_exchange_rate method take in a FromCurrency and a ToCurrency that extends Currency.

Reply:
Maybe this is just a naming problem? Since Currency does not imply the CurrencyFromTo or whatever we are doing here.

Reply:
By the way remove 
+/**
+* Interval parameter, i.e interval=15min
+*/

Reply:
Btw, I think to have a inheritance chain right now maybe is a bit redundant, I think the simplest thing would be to have FromCurrency and ToCurrency and skip the inheritance. The reuse of one method for two classes creates more code than it gives reuse. What do you think @Braisly?

Reply:
I think that the way to be simple and understandable is to create two files how @patriques82 was saying. Also it could be done renaming the file but I will create 2 files to be less complex.